### PR TITLE
Upgrading to Cordova 4.4.0 and switching to dynamic framework targets

### PR DESCRIPTION
**Pull request by bhariharan**
This PR upgrades `Cordova` to `4.4.0` and updates `SalesforceHybridSDK` and sample apps' consumption of it to the dynamic framework target instead of the static target.

### CODE REVIEW

**Comment by bhariharan** [comment]
@sgoldberg-sfdc FYI

**Comment by huminzhi** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Code coverage under minimum of 80%</td>
    </tr>
  </tbody>
</table>

## Current coverage for SmartSync.framework is `73%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceAnalytics.framework is `41%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceSDKCore.framework is `22%`
✅ *No files affecting coverage found*

---
## Current coverage for SmartStore.framework is `38%`
✅ *No files affecting coverage found*

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -891,11 +892,6 @@
 			isa = PBXGroup;
 			children = (
 				CE2B8CFB1CE9322D00C6FC6A /* SalesforceAnalytics.framework */,
-				CEA884851C191F60008D871B /* SalesforceSDKCore.framework */,
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj]: There were 2 dependencies. I removed the duplicate entry.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1370,6 +1366,13 @@
 			remoteRef = CEA8844E1C191796008D871B /* PBXContainerItemProxy */;
 			sourceTree = BUILT_PRODUCTS_DIR;
 		};
+		CEEF966C1EB7CBF50059CA79 /* Cordova.framework */ = {
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj]: The static target uses Cordova's static target and the dynamic target uses Cordova's dynamic target.

**Pull request merged**